jerry-debugger/src/lib/cdt-proxy.ts

```diff
@@ -36,13 +44,62 @@ export class ChromeDevToolsProxyServer {
       ws.send('hello');
     });
 
+    server.listen(SERVER_PORT);
+
+    // TODO: move callback out to standalone method
```

Comment:
Maybe even a new file?

Could do sth like:
```
// cdt-proxy-http.ts

import { ChromeDevToolsProxyServer } from './cdt-proxy';
// other imports you need here

export function onHttpRequest(this: ChromeDevToolsProxyServer,
                              request: http.IncomingMessage, response: http.ServerResponse) {
  ... // the code you've got here
}

// cdt-proxy.ts
...
import { onHttpRequest } from './cdt-proxy-http';

  server.on('request', onHttpRequest.bind(this));
```

Splitting also makes unit testing much easier.

Speaking of testing... As soon as a piece of code gets beyond "prototyping / playing around", I'd like to see some unit tests for that piece of code. I'm afraid that if writing tests doesn't happen along the way, we'll end up with hard-to-test code in the end.

Reply:
Wow, a circular dependency between the two files? This is pretty unintuitive to me, I try to avoid such things. :) Help me understand better why you would do that in this case; you don't want to pull every function out into its own file I presume. I'm unsure about how to do the testing, so maybe that's the reasoning and I don't get it.

Reply:
Good point about the circular dependency. I think it's OKish because it's only used for type-checking `this` (i.e. it's not used used as a constructor).

There are cleaner ways (i.e. have a separate file with the common interface and make the class implement it) but I wouldn't worry too much about it for now, TBH.

> I'm unsure about how to do the testing, so maybe that's the reasoning and I don't get it.

Do you mean you're unsure about how to use the test framework, or how to approach (unit) testing in general, or..? Let's talk about this in our next sync up call.